flang/include/flang/Optimizer/Dialect/FIROps.td

```diff
@@ -2106,7 +2106,7 @@ def fir_DoLoopOp : region_Op<"do_loop",
     mlir::OpBuilder getBodyBuilder() {
       return mlir::OpBuilder(getBody(), std::prev(getBody()->end()));
     }
-    mlir::Block::BlockArgListType getRegionIterArgs() {
```

Comment:
Flang loops have some special semantics around iter_args and I'm not sure how to expose inits, iter_args and yielded values through the interface in a way that it verifies. (I think it's possible, but by renaming this function I can skip this verification for now; exposing iter_args through the interface is optional.)